provider_keep_alive_hook/use_provider_keep_alive.dart

```diff
@@ -1,77 +1,53 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-/// A hook that prevents a provider from being automatically disposed.
-///
-/// This hook listens to a provider (similarly to [useProvider]), without triggering a rebuild when the provider value changes.
-///
-/// Example:
-///
-/// ```dart
-/// class Example extends HookWidget {
-///   @override
-///   Widget build(BuildContext context) {
-///     useProviderKeepAlive(counterProvider);
-///     return Container();
-///   }
-/// }
-/// ```
+/// Hook that returns the provider and calls [onInit] exactly once.
 ///
-/// As long as the `Example` widget is in the widget tree, the `counterProvider` will not get disposed.
-/// When the `counterProvider` value changes the `Example` widget will not get rebuild.
-void useProviderKeepAlive(ProviderListenable provider) {
-  final container = ProviderScope.containerOf(useContext());
-  use<void>(_ProviderKeepAliveHook(container, provider));
+/// This is designed to be used with init method on provider to initialize some runtime dependencies.
+T useInitProvider<T>(WidgetRef ref, ProviderListenable<T> provider, Function(T) onInit) {
```

Comment:
1. Are you sure you are editing the right hook. The purpose of this hook is to keep a provider alive without listening to it, and not to init a provider. Did you mean to edit the `/init_provider_hook/use_init_provider.dart`?
2. The implementation in `/init_provider_hook/use_init_provider.dart` might be better than this implementation.
3. Do we even need a `useInitProvider`? We could have a general `useInit(void Function() callback)` hook. We would use it like:
```dart
Widget build(BuildContext context){
    final provider = context.watch(myProvider);
    useInit((){
        provider.init();
    });
    // ...
}
```

Reply:
2. Ok, I guess we need to pass the `WidgetRef`. But why not use the `useEffect` here:
```dart
T useInitProvider<T>(WidgetRef ref, ProviderListenable<T> provider, Function(T) onInit) {
  final _provider = ref.watch(provider);
  useEffect(() {
    onInit(_provider);
    return null;
  }, []);
  return _provider;
}
``` 
3. I know this can be done with `useEffect`, but I would still consider having  the `useInit` hook. It provides a bit more meaning, and we don't have to return `null`.

```dart
void useInit(void Function() callback) {
  useEffect(() {
    callback();
    return null;
  }, []);
}
```